### DISCUSSION TITLE
[CLIENT] Using directives on TopLevelObjects

### DISCUSSION THREAD

**Discussion by FidasL** [ExpediaGroup/graphql-kotlin#1281] [Q&A] [answered]
Hi!
I have custom graphql-server and graphql-client both developed with `graphql-kotlin` library. On the client side, I want to use schema directives on topLevelObject:
```
query {
  students @include(if: false) {
    id
    firstName
    lastName
  }
}
```

But after execute query like this from client, request serialization fails, because it generates non-nullable types in `Result` class:
```
  public data class Result(
    public val students: List<Student> // -> this type should be nullable
  )
```

Please, is this an issue, or do I need to configure something differently? Our client version is `4.1.1`. Thank you :)

**Comment by dariuszkuc** [comment, marked as answer]
Hello 👋 
Unfortunately we currently don't support query directives in the client. I've opened up #1282 to track it.
